src/Corax/Querying/Matches/TermProviders/TermProvider.NumericRange.cs

```diff
@@ -59,56 +59,74 @@ false when typeof(TLow) == typeof(Range.Inclusive) && typeof(TVal) == typeof(Dou
             PrepareKeys();
             Reset();
         }
-
+        
         private void PrepareKeys()
         {
+            // We want to rewrite the range, so it starts and ends with elements that are presented in our data source
             TVal startKey = _iterator.IsForward ? _low : _high;
             TVal finalKey = _iterator.IsForward ? _high : _low;
 
+            // Seek for startKey in the iterator
             _iterator.Seek(startKey);
-            if (_iterator.MoveNext(out TVal key, out _, out _) == false)
+            
+            // We get either
+            // currentKey equal to startKey
+            // currentKey equal to element succeeding startKey
+            // MoveNext returns false
+            if (_iterator.MoveNext(out TVal currentKey, out _, out _) == false)
             {
+                // When MoveNext returns false it means we jumped over all values stored inside a lookup tree
                 _isEmpty = true;
                 return;
             }
 
-            var skipFirst = _iterator.IsForward switch
-            {
-                true when typeof(TLow) == typeof(Range.Exclusive) && key.IsEqual(_low) => true,
-                false when typeof(THigh) == typeof(Range.Exclusive) && _high.CompareTo(key) <= 0 => true,
-                false when typeof(THigh) == typeof(Range.Inclusive) && _high.CompareTo(key) < 0 => true,
-                _ => false
-            };
-
+            // The first element may be equal to the startKey. For exclusive range start we want to skip it
+            var skipFirst = (_iterator.IsForward ? typeof(TLow) : typeof(THigh)) == typeof(Range.Exclusive) 
+                            && startKey.IsEqual(currentKey);
+            
+            // If we're supposed to skip first element, we rewrite the range to start from element succeeding startKey
             if (skipFirst)
             {
-                if (_iterator.MoveNext(out key, out _, out _) == false)
+                if (_iterator.MoveNext(out currentKey, out _, out _) == false)
                 {
+                    // We moved past queried range, nothing matches the query
                     _isEmpty = true;
                     return;
                 }
-                
-                if (_iterator.IsForward)
-                    _low = key;
-                else
-                    _high = key;
             }
             
+            // Update the range accordingly to the iterator option
+            if (_iterator.IsForward)
+                _low = currentKey;
+            else
+                _high = currentKey;
+            
             if (_skipRangeCheck)
-                return; // to the end
+                return;
             
-            //Now seek to the last key
+            // Seek for finalKey in the iterator
             _iterator.Seek(finalKey);
```

Comment:
In comments explain what you are doing and more importantly why. Avoid using as much as possible the variables names, because those are easy to miss when you change the code and comments end up being inconsistent.

Reply:
I think we don't need any comment here since `Seek` method is commented now

Reply:
Probably in this case it is self explanatory but in general comments in methods are about guarantees, requirements and expectations on the method and the caller. On the caller code itself is about why that call is there, what's the purpose of it, what side effects are expected, etc.